fe/fe-core/src/main/java/org/apache/doris/transaction/DatabaseTransactionMgr.java

```diff
@@ -2830,13 +2830,24 @@ private PublishResult checkQuorumReplicas(TransactionState transactionState, lon
     }
 
     protected void addSubTransaction(long transactionId, long subTransactionId) {
-        subTxnIdToTxnId.put(subTransactionId, transactionId);
+        writeLock();
+        try {
+            subTxnIdToTxnId.put(subTransactionId, transactionId);
```

Comment:
`subTxnIdToTxnId` is a concurrent map, is it necessary to lock it?

Reply:
it's not concurrent map `private final Map<Long, Long> subTxnIdToTxnId = Maps.newHashMap()`